dataclients/kubernetes/token_source.go

```diff
@@ -0,0 +1,26 @@
+package kubernetes
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"golang.org/x/oauth2"
+)
+
+type fileTokenSource struct {
+	path   string
+	period time.Duration
+}
+
+func (ts *fileTokenSource) Token() (*oauth2.Token, error) {
+	token, err := ioutil.ReadFile(ts.path)
+	if err != nil {
```

Comment:
in case of IO errors, would it make sense to return the previous token? Or is this handled by oauth2.ReuseTokenSource?

Reply:
discussed IRL, it's fine